server/application-server/src/main/java/de/tum/in/www1/hephaestus/codereview/user/UserRepository.java

```diff
@@ -0,0 +1,29 @@
+package de.tum.in.www1.hephaestus.codereview.user;
+
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+
+    @Query("SELECT u FROM User u WHERE u.login = :login")
+    Optional<User> findUser(@Param("login") String login);
+
+    @Query("""
+                SELECT new de.tum.in.www1.hephaestus.codereview.user.UserDTO(u.login, u.email, u.name, u.url)
+                FROM User u
+                WHERE u.login = :login
+            """)
+    Optional<UserDTO> findByLogin(@Param("login") String login);
```

Comment:
Bonus to the comment above that we should anyways do, can you try removing the `@Query` and see if it still works? I think since the `UserDTO` is a subset of `User` without type conversions this should automatically be handled by hibernate.

Reply:
In theory it would work, but only for the full arg constructors as it seems. Since here we want to use the smaller constructor, it doesn't work (just tested it).

Reply:
Good to know, thanks!